pyanaconda/modules/subscription/subscription.py

```diff
@@ -556,3 +559,48 @@ def set_rhsm_config_with_task(self):
                                         rhsm_config_defaults=self.get_rhsm_config_defaults(),
                                         subscription_request=self._subscription_request)
         return task
+
+    def register_username_password_with_task(self):
+        """Register with username and password based on current subscription request.
+
+        :return: a DBus path of an installation task
+        """
+        # NOTE: we access self._subscription_request directly
+        #       to avoid the sensitive data clearing happening
+        #       in the subscription_request property getter
+        username = self._subscription_request.account_username
+        password = self._subscription_request.account_password.value
+        register_server_proxy = self.rhsm_observer.get_proxy(RHSM_REGISTER_SERVER)
+        task = RegisterWithUsernamePasswordTask(rhsm_register_server_proxy=register_server_proxy,
+                                                username=username,
+                                                password=password)
+        return task
+
+    def register_organization_key_with_task(self):
+        """Register with organization and activation key(s) based on current subscription request.
+
+        :return: a DBus path of an installation task
+        """
+        # NOTE: we access self._subscription_request directly
+        #       to avoid the sensitive data clearing happening
+        #       in the subscription_request property getter
+        organization = self._subscription_request.organization
+        activation_keys = self._subscription_request.activation_keys.value
+        register_server_proxy = self.rhsm_observer.get_proxy(RHSM_REGISTER_SERVER)
+        task = RegisterWithOrganizationKeyTask(rhsm_register_server_proxy=register_server_proxy,
+                                               organization=organization,
+                                               activation_keys=activation_keys)
+        return task
+
+    def unregister_with_task(self):
+        """Unregister the system.
+
+        :return: a DBus path of an installation task
+        """
+        rhsm_unregister_proxy = self.rhsm_observer.get_proxy(RHSM_UNREGISTER)
+        task = UnregisterTask(rhsm_unregister_proxy=rhsm_unregister_proxy)
+        # we will no longer be registered and subscribed inf the task is successful,
+        # so set the corresponding property appropriately
+        task.succeeded_signal.connect(
+            lambda: self.set_subscription_attached(False))
```

Comment:
I don't think that this value is ever set to True.

Reply:
Yet - the soon-to-be-added task that attaches subscription will set it to True after successful run.